waku/v2/discv5/discover_test.go

```diff
@@ -15,6 +20,51 @@ import (
 	"github.com/waku-org/go-waku/waku/v2/utils"
 )
 
+func discoverFilterOnDemand(iterator enode.Iterator, maxCount int) ([]service.PeerData, error) {
```

Comment:
This should already be covered in peer manager tests as https://github.com/waku-org/go-waku/blob/826c7fb924e1f070e414773d9bee76998ad479aa/waku/v2/peermanager/peer_discovery.go#L52-L103 is already doing same functionality. 

Is it not being show in coverage report?

Reply:
Yes, the main reason I come up with test cases for FilterShard() and FilterCapabilities() was they didn't show up as covered. Another reason why to test code which is already used somewhere else is the absence of any assertions. Unit tests serve also as a guard for regressions or could detect race conditions after unit got affected by changes somewhere else. I agree best value would be to come up with completely unique tests. This will eventually come later once I learn and seal the main flow.

Reply:
Odd that they are not showing as covered, maybe the coverage report is not aggregating coverage from other tests is it? I thought i fixed this at some point.
Yeah, i just don;t want soo much code to be written to simulate things as it would become more maintenance prone as we need to change something especially if it is getting covered by some other test in another package.

Reply:
I have checked again:
1) Commented out my tests TestDiscV5WithShardFilter and TestDiscV5WithCapabilitiesFilter 
2) Run coverage report at /go-waku dir with "go test -coverprofile=coverage.out ./..."
3) FilterShard and FilterCapabilities haven't show up as covered

Looked up with GPT, tests from other packages could cover funcs in "local" package if:
1) There is explicit import and use of "remote" package
2) "go test -coverprofile=coverage.out ./..."   is used

discoverOnDemand() in peermanager package is not public, so it cannot be explicitly imported. And it is a normal code without any assertions. Even I believe is works well and no issues about it, I still need some code to build my test case to cover  FilterShard and FilterCapabilities. If we agree we would like to have better coverage which is traceable. 

I got your point, it really is more maintenance work. Coverage tests create sort of a cage limiting freedom of devs to be that fast. I briefly discussed with @fbarbu15 wether QA team could maintain tests broken with new changes and he agreed on it. I personally like to learn along the changes you devs introduce, so no problem on my side either. We would just need to hand over PRs from dev to QA and back. Please reach to you leaders if this model is we want. We should ideally get both quality and still deliver changes at decent speed?

Reply:
Ah..Thanks for finding it out.

Ok, in that case then it is probably ok to proceed with this.